File: perl/version1.pl
#!/usr/bin/perl
# Version 1: Consume SOAP service directly
# Usage: perl version1.pl <number>
# Example: perl version1.pl 10

use strict;
use warnings;
use SOAP::Lite;

# Get number from command line argument
my $number = $ARGV[0] || '10';

# Configure SOAP client
my $wsdl = 'https://www.dataaccess.com/webservicesserver/NumberConversion.wso?WSDL';
my $client = SOAP::Lite->service($wsdl);

# Call the NumberToWords operation
my $result = $client->NumberToWords($number);

# Display the result
print "$result\n";

File: ruby/version1.rb
#!/usr/bin/env ruby
# Version 1: Consume SOAP service directly
# Usage: ruby version1.rb <number>
# Example: ruby version1.rb 10

require 'savon'

# Get number from command line argument
number = ARGV[0] || '10'

# Configure SOAP client
client = Savon.client(
  wsdl: 'https://www.dataaccess.com/webservicesserver/NumberConversion.wso?WSDL',
  log: false
)

# Call the NumberToWords operation
response = client.call(
  :number_to_words,
  message: { 'ubiNum' => number }
)

# Extract and display the result
result = response.body[:number_to_words_response][:number_to_words_result]
puts result
